src/components/KeyboardShortcuts.tsx

```diff
@@ -1,51 +1,95 @@
 import { useEffect } from 'preact/hooks'
-import { useThemeStore, useUtilityBarStore } from '../features/utility-bar/store'
+import {
+  useThemeStore,
+  useUtilityBarStore,
+} from '../features/utility-bar/store'
+import { usePostMessageReceiver } from '../features/iframe-postmessage/receiver'
+import { isKeyboardEventMessage } from '../features/iframe-postmessage/messages'
+
+/**
+ * A simplified version of a keyboard event.
+ * preventDefault is optional so we can safely
+ * serialize it in a postMessage
+ */
+export interface SimpleKeyboardEvent {
+  key: string
+  shiftKey: boolean
+  preventDefault?(): void
+}
+
+export function canHandleKeyboard(): boolean {
+  return !(
+    document.activeElement instanceof HTMLInputElement ||
+    document.activeElement instanceof HTMLTextAreaElement
+  )
+}
+
+/**
+ * Safely preventDefault when we can
+ */
+function preventDefault(e: SimpleKeyboardEvent) {
+  if (e.preventDefault !== undefined) {
+    e.preventDefault()
+  }
+}
 
 export default function () {
   const utilityStore = useUtilityBarStore()
   const themeStore = useThemeStore()
 
+  usePostMessageReceiver({
+    onMessage(e) {
+      if (!isKeyboardEventMessage(e.data)) {
+        return
+      }
+
+      keydownHandler(e.data.payload)
+    },
+  })
+
   // handles different keyboard shortcuts
-  const keydownHandler = (e: KeyboardEvent) => {
-    // don't assess shortcuts when inside of an input or textarea
-    if (
-      document.activeElement instanceof HTMLInputElement ||
-      document.activeElement instanceof HTMLTextAreaElement
-    ) {
+  const keydownHandler = (e: SimpleKeyboardEvent) => {
+    if (!canHandleKeyboard()) {
       return
     }
 
+    console.log({
+      key: e.key,
+      shiftKey: e.shiftKey,
+      preventDefault: e.preventDefault
+    })
+
```

Comment:
This was nice for testing purposes but you may want to yoink this out before merge.

Reply:
Good catch!